### PR TITLE
Unable to deleted the root site

### LINKED ISSUE #202: Root site can be deleted
Result is white screen, unrecoverable.

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #202

### CODE REVIEW

**Comment by sbwalker** [comment]
Instead of adding a Disabled property to the ActionDialog, can you add the validation logic to the Index.razor for the Site.

**Comment by hishamco** [comment]
I was planned to do that first, but I see value to have such property in `ActionDialog` coz there 're some case to have the `ActionLink` disabled

**Comment by sbwalker** [comment]
Good point! Can you also add this feature to ActionLink as well? Also there is a merge conflict in ActionDialog.

**Comment by hishamco** [comment]
Ok, I will handle this ASAP

**Comment by sbwalker** [comment]
Thank for addressing the merge conflict

File: Oqtane.Server/Repository/SiteRepository.cs [hunk 1]
```diff
@@ -142,10 +142,13 @@ public Site GetSite(int siteId)
         }
 
         public void DeleteSite(int siteId)
-        {
-            Site site = db.Site.Find(siteId);
-            db.Site.Remove(site);
-            db.SaveChanges();
+        {       
+            if (db.Site.Count() > 1)
```
Comment (chlupac) [inline Oqtane.Server/Repository/SiteRepository.cs]: Site.Count is not enough. Root(first) site have to be protected. Strange things happen when you do it.

**Pull request merged**